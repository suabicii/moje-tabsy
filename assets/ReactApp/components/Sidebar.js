import React from "react";
import {NavLink} from "react-router-dom";
import {mainRoute} from "../routers/AppRouter";

function Sidebar() {
    return (
        // Sidebar -->
        <nav id="sidebarMenu" className="collapse d-lg-block sidebar collapse bg-white">
            <div className="position-sticky">
                <div className="list-group list-group-flush mx-3 mt-4">
                    <NavLink to={mainRoute} className="list-group-item list-group-item-action py-2 ripple" end>
                        <i className="fas fa-tachometer-alt fa-fw me-3"></i><span>Strona główna</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/drug-list`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-jar fa-fw me-3"></i><span>Moje leki i suplementy</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/profile`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-user fa-fw me-3"></i><span>Mój profil</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/settings`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-gear fa-fw me-3"></i><span>Ustawienia</span>
                    </NavLink>
                    <a href="/logout" className="list-group-item list-group-item-action py-2 ripple"
                    ><i className="fa-solid fa-power-off fa-fw me-3"></i><span>Wyloguj się</span></a
                    >
                </div>
            </div>
        </nav>
        //Sidebar -->
    );
}

export default Sidebar;