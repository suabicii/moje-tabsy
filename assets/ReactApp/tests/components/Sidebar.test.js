import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Sidebar from "../../components/Sidebar";

it('should correctly render sidebar', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Sidebar/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});