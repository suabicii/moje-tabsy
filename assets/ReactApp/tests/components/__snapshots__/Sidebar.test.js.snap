// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render sidebar 1`] = `
<nav
  className="collapse d-lg-block sidebar collapse bg-white"
  id="sidebarMenu"
>
  <div
    className="position-sticky"
  >
    <div
      className="list-group list-group-flush mx-3 mt-4"
    >
      <NavLink
        className="list-group-item list-group-item-action py-2 ripple"
        end={true}
        to="/dashboard"
      >
        <i
          className="fas fa-tachometer-alt fa-fw me-3"
        />
        <span>
          Strona główna
        </span>
      </NavLink>
      <NavLink
        className="list-group-item list-group-item-action py-2 ripple"
        to="/dashboard/drug-list"
      >
        <i
          className="fa-solid fa-jar fa-fw me-3"
        />
        <span>
          Moje leki i suplementy
        </span>
      </NavLink>
      <NavLink
        className="list-group-item list-group-item-action py-2 ripple"
        to="/dashboard/profile"
      >
        <i
          className="fa-solid fa-user fa-fw me-3"
        />
        <span>
          Mój profil
        </span>
      </NavLink>
      <NavLink
        className="list-group-item list-group-item-action py-2 ripple"
        to="/dashboard/settings"
      >
        <i
          className="fa-solid fa-gear fa-fw me-3"
        />
        <span>
          Ustawienia
        </span>
      </NavLink>
      <a
        className="list-group-item list-group-item-action py-2 ripple"
        href="/logout"
      >
        <i
          className="fa-solid fa-power-off fa-fw me-3"
        />
        <span>
          Wyloguj się
        </span>
      </a>
    </div>
  </div>
</nav>
`;
